hyperspy/io_plugins/nexus.py

```diff
@@ -0,0 +1,1251 @@
+def _guess_signal_type(data):   
+    """Checks last 2 dimensions
+    An area detector will be square, > 255 pixels each side and
+    if rectangular ratio of sides is at most 4.
+    An EDS detector will generally by ndetectors*2048 or ndetectors*4096
+    the ratio test will be X10-1000  
+    
+    Parameters
+    ----------
+    data : dask or numpy array
+    
+    Returns
+    -------
+    signal : Hyperspy Signal
+        Estimate of signal type - "Signal1D", "Signal2D" or "BaseSignal"
+    
+    """
+    if len(data.shape) == 1:
+        signal_type = "Signal1D"
+    elif len(data.shape) == 2:
+        signal_type = "Signal2D"
+    else:
+        n1 = data.shape[-1]
+        n2 = data.shape[-2]
+        if n2>127 and n1 >127 and n1/n2 < 1.5:
+            signal_type = "Signal2D"
+        elif n1 > 127 and n2 < n1:
+            signal_type = "Signal1D"
+        else:
+            signal_type = "BaseSignal"
+    return signal_type
+        
+
+
+def read_metadata_from_file(filename,search_keys="all",
+                        small_metadata_only=False,
+                        verbose=False):   
+    """ Read the metadata from a nexus or hdf file       
+    The method iterates through the group and returns a dictionary of 
+    the entries.
+    This is a convenience method to inspect a file for a value
+    rather than loading the file as a signal
+    
+    Parameters
+    ----------
+    filename : str  
+        path of the file to read
+    search_keys  : {'all',str or list_of_strings} , default : 'all'
+        'all' will return all datasets found including linked data
+        A list of strings will only return items which contain the strings
+        For example, search_keys = ["instrument","Fe"] will return
+        hdf entries with instrument or Fe in their hdf path.
+    verbose: bool, default : False  
+        Pretty Print the results to screen 
+
+
+    See Also 
+    --------
+    * :py:meth:`~.io_plugins.nexus.file_reader`
+    * :py:meth:`~.io_plugins.nexus.file_writer`
+    * :py:meth:`~.io_plugins.nexus.read_datasets_from_file`
+    
+    
+    Returns
+    -------
+    dict
+        Metadata dictionary. When search_list is specified only items
+        matching the search keys will be returned 
+        
+    """
+    search_keys = _check_search_keys(search_keys)
+    fin = h5py.File(filename,"r")
+    # search for NXdata sets...
+    # strip out the metadata (basically everything other than NXdata)
+    stripped_metadata = _load_metadata(fin,search_keys=search_keys)
+    # strip out the search keys
+    #stripped_metadata = _find_search_keys_in_dict(metadata,search_keys=search_keys)
+    # strip out the small metadata if needed
+    if small_metadata_only:
+        stripped_metadata = _find_smalldata_in_dict(stripped_metadata)
+    if verbose:
+        pprint.pprint(stripped_metadata)
+    return stripped_metadata    
+
+
+def read_datasets_from_file(filename,search_keys=None,
+                          verbose=True):
```

Comment:
Thanks

The empty_list was due to the default setting - nxdata_only =True. This isn't generic enough so I've changed to nxdata_only = False default. 
I've found the error which is due to incorrect formatting in the file but I've put in a check if
its a str instead of a list.

I've tested against the following files from the nexus examples so think I've covered al variations.
DLS\i16\hdf5\538039.nxs
DLS\NXquadric\hdf5\sample_capillary.nxs
DLS\p45\hdf5\p45-316.nxs
DLS\p45\hdf5\p45-1168.nxs
DLS\p45\hdf5\p45-2194.nxs
DLS\reflections\hdf5\thaumatin_integrated.nxs
DLS\reflections\hdf5\thaumatin_integrated_multisample.nxs
Soleil\hdf5\file_1.nxsSoleil\hdf5\file_2.nxs
SwissFEL\hdf5\lyso009a_0087.JF07T32V01_master.h5
nxpdb\hdf5\4n8z.h5   - this hs string arrays only  returns empty
IPNS\LRMECS\hdf5\lrcs3701.nx5
hdf5\writer_1_3__niac2014.h5
hdf5\focus2007n001335.hdf
hdf5\NXtest.h5
APS\scan2nexus\hdf4\2iddf_0106.nexus
APS\scan2nexus\hdf4\mts_0347.nexus
APS\EPICSareaDetector\hdf5\AgBehenate_228.hdf5
APS\NXsas\hdf5\nexus-example.hdf5
DLS\i03_i04_NXmx\hdf5\Therm_6_2.nxs -   this previously failed in your test
DLS\p45\hdf5\p45-2194.nxs
DLS\p45\hdf5\p45-316.nxs